featurebyte/routes/context/controller.py

```diff
@@ -42,6 +44,42 @@ def __init__(
         self.entity_service = entity_service
         self.use_case_service = use_case_service
         self.catalog_service = catalog_service
+        self.derive_primary_entity_helper = derive_primary_entity_helper
+
+    async def create_context(
+        self,
+        data: ContextCreate,
+    ) -> ContextModel:
+        """
+        Create Context
+
+        Parameters
+        ----------
+        data: ContextCreate
+            Context creation payload
+
+        Raises
+        ------
+        DocumentCreationError
+            raise when context entity ids are not all primary entity ids
+
+        Returns
+        -------
+        ContextModel
+        """
+        # validate entity ids exist
+        for entity_id in data.primary_entity_ids:
+            await self.entity_service.get_document(document_id=entity_id)
+
+        # validate all entity ids are primary entity ids
+        primary_entity_ids = await self.derive_primary_entity_helper.derive_primary_entity_ids(
+            entity_ids=data.primary_entity_ids
+        )
+        if set(primary_entity_ids) != set(data.primary_entity_ids):
+            raise DocumentCreationError("Context entity ids must all be primary entity ids")
```

Comment:
can consider adding some more information in the message. might be useful to include which entities are that ones that are not primary entities

Reply:
will try to add later in another PR